pkg/view/manager.go

```diff
@@ -0,0 +1,132 @@
+package view
+
+import (
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-faster/errors"
+	"github.com/rs/zerolog/log"
+	"html/template"
+	"io/fs"
+	"net/http"
+	"path/filepath"
+)
+
+type TemplateHandler func(w http.ResponseWriter, r *http.Request) any
+
+type TemplateRoute struct {
+	name    string
+	handler func(w http.ResponseWriter, r *http.Request) any
+	tmpl    *template.Template
+}
+
+type Manager struct {
+	templates map[string]*TemplateRoute
+	tfs       fs.FS
+	watcher   *fsnotify.Watcher
+}
+
+func NewManager(fsys fs.FS) (*Manager, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create file watcher")
+	}
+
+	m := &Manager{
+		templates: make(map[string]*TemplateRoute),
+		tfs:       fsys,
+		watcher:   watcher,
+	}
+	err = m.loadTemplates()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load templates")
+	}
+
+	// TODO breadchris setup fs different in dev for live reload
+	//go m.watchTemplateChanges()
```

Comment:
i'm fine with a config option that reads templates from a local filesystem for local dev and uses go:embed for production.